src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -650,3 +652,141 @@ def _allocate_col(
         .rename(columns={output_col: allocate_col})
     )
     return to_allocate.loc[:, [allocate_col]]
+
+
+##################
+# Validation Tests
+##################
+
+# This whole section should pprrrrooobably be moved into a validation layer of
+# CI tests for this repo.
+
+def test_consistency_of_data_stages(df1, df2):
+    """
+    Test the consistency of two stages of the depreciation data processing.
+
+    The data that is processed in this repo goes along multiple stages of its
+    journey. This function right now is hard coded to test two the depreication
+    data's main data columns. Right now, this is hard coded to fail when there
+    are more inconsitent plants and utilities than are currently known.
+
+    TODO: I think :func:``data_col_test`` is a perfect candidate for the
+    decorators that let you test multiple inputs (different stages of the data
+    and different data columns) as well as output values (in this case the
+    number of known bad utilities and plants).
+
+    Args:
+        df1: One dataframe to sum and check consistency with ``df2``. Both
+            dfs should have columns ``data_col`` as well as identifying
+            columns: ``['report_date', 'data_source', 'utility_id_pudl',
+            'plant_id_eia']``
+        df2: Other dataframe to sum and check consistency against ``df1``.
+
+    """
+    util_bad, plant_bad = data_col_test(
+        df1=df1, df2=df2, data_col='plant_balance_w_common')
+
+    # known baddies. We need to track these down
+    assert(len(plant_bad) <= 53)
+    assert(len(util_bad) <= 88)
+
+    util_bad, plant_bad = data_col_test(
+        df1=df1, df2=df2, data_col='unaccrued_balance_w_common')
+
+    # known baddies. We need to track these down
+    assert(len(plant_bad) <= 35)
+    assert(len(util_bad) <= 70)
+
+
+def data_col_test(df1, df2, data_col: str) -> pd.DataFrame:
```

Comment:
You might want to give df1 and df2 types as well, and I think the return type is incorrect here -- you're returning a tuple of DataFrames.